src/plugins/unified_search/public/filter_badge/filter_badge.tsx

```diff
@@ -53,7 +55,7 @@ function FilterBadge({
     <EuiBadge
       className={badgePaddingCss(euiTheme)}
       color="hollow"
-      iconType="cross"
+      iconType={readOnly ? 'cross' : undefined}
```

Comment:
@elastic/kibana-visualizations The only change I made here is that the `x` icon doesn't show on the filter pill when in read only mode.
<img width="445" alt="Screenshot 2023-08-14 at 10 52 05 AM" src="https://github.com/elastic/kibana/assets/1697105/e10a791f-da9b-4522-9c3f-980aa813688b">